### DISCUSSION TITLE
libs/header/sync: Consider timeout on header requests via exchange

### DISCUSSION THREAD

**Discussion by renaynay** [celestiaorg/celestia-node#3045] [Ideas]
Exchange request sessions can hang as long as caller's context allows (or as long as exchange is running) -- consider implementing a timeout on syncer-side to prevent a hanging request in the case 1 header is not served for some reason (as unlikely as that is).

**Comment by Wondertan** [comment]
I really doubt we need the context in the syncer when it accesses exchange. It's the same problem as for DASer and GetByHeight. There is really nothing Syncer can do besides waiting until Exchange gives the requested header, and it's the responsibility of the exchange to retry internally
